Maya/Exporter/BabylonExporter.GLTFExporter.Material.cs

```diff
@@ -562,47 +562,69 @@ private bool isTextureOk(BabylonTexture texture)
-                RaiseVerbose("metallic=" + metallic, 3);
-                RaiseVerbose("diffuseScaleFactor=" + diffuseScaleFactor, 3);
-                RaiseVerbose("baseColorFromDiffuse=" + baseColorFromDiffuse, 3);
-                RaiseVerbose("baseColorFromSpecular=" + baseColorFromSpecular, 3);
-                RaiseVerbose("metallic * metallic=" + metallic * metallic, 3);
-                RaiseVerbose("baseColor=" + baseColor, 3);
-                RaiseVerbose("-----------------------", 3);
-            }
-
-            return new MetallicRoughness
-            {
-                baseColor = baseColor,
-                opacity = opacity,
-                metallic = metallic,
-                roughness = 1 - glossiness
-            };
+        private MetallicRoughness ConvertToMetallicRoughness(SpecularGlossiness specularGlossiness, bool displayPrints = false)
+        {
+            // Hard coded points used to define the specular power to roughness curve.
+            var P0 = new BabylonVector2(0f, 1f);
+            var P1 = new BabylonVector2(0f, 0.1f);
+            var P2 = new BabylonVector2(0f, 0.1f);
+            var P3 = new BabylonVector2(1300f, 0.1f);
+
+            /**
+             * Helper function that defines the bezier curve as well.Given the control points, solve for x based on a given t for a cubic bezier curve
+             * @param t a value between 0 and 1
+             * @param p0 first control point
+             * @param p1 second control point
+             * @param p2 third control point
+             * @param p3 fourth control point
+             * @returns number result of cubic bezier curve at the specified t
+             */
+            float _cubicBezierCurve(float t, float p0, float p1, float p2, float p3)
+            {
+                return
+                (
+                    (1 - t) * (1 - t) * (1 - t) * p0 +
+                    3 * (1 - t) * (1 - t) * t * p1 +
+                    3 * (1 - t) * t * t * p2 +
+                    t * t * t * p3
+                );
+            }
+
+            /*
+             * Helper function that calculates a roughness coefficient given a blinn-phong specular power coefficient
+             * @param specularPower the blinn-phong specular power coefficient
+             * @returns number result of specularPower -> roughness conversion curve.
+             */
+            float _solveForRoughness(float specularPower)
+            {
+                var t = Math.Pow(specularPower / P3.X, 0.333333);
+                return _cubicBezierCurve((float)t, P0.Y, P1.Y, P2.Y, P3.Y);
+            }
+
+            var diffuse = specularGlossiness.diffuse.scale(0.5f);
+            var opacity = specularGlossiness.opacity;
+            var glossiness = specularGlossiness.glossiness;
+            var metallic = 0;
+            var roughness = _solveForRoughness(glossiness * 256); // Glossiness = specularPower / 256

```

Comment:
Where does 256 come from?

Reply:
When we construct our SpecularGlossiness structure from the babylonMaterial we converted, we divide the specularPower by 256. I'm not entirely sure why...

https://github.com/BabylonJS/Exporters/blob/b45ec95831d630faff22f291192d4be542a11330/3ds%20Max/Max2Babylon/Exporter/BabylonExporter.GLTFExporter.Material.cs#L186

Reply:
But this is not a Babylon material right?

Reply:
and when we initially convert from Max->Babylon we, we multiply the shininess (Gloss in Max) by 256 to become what we define as SpecularPower in Babylon...strange...

Looking at the source, the shininess of a material is represented in max as a 0-1 value, while we represent it in Babylon as a 0-infinity value. I'm not sure why we use a factor of 256 other than to get this "looking right".

https://github.com/BabylonJS/Exporters/blob/b45ec95831d630faff22f291192d4be542a11330/3ds%20Max/Max2Babylon/Exporter/BabylonExporter.Material.cs#L152